service-registry/src/main/java/org/apache/servicecomb/serviceregistry/task/MicroserviceRegisterTask.java

```diff
@@ -155,7 +155,7 @@ private boolean registerSchemas() {
         String curSchemaSumary = existSchema.getSummary();
         String schemaSummary = Hashing.sha256().newHasher().putString(content, Charsets.UTF_8).hash().toString();
         if (!schemaSummary.equals(curSchemaSumary)) {
-          if (microservice.getInstance().getEnvironment().equalsIgnoreCase("development")) {
+          if (curSchemaSumary == null || microservice.getInstance().getEnvironment().equalsIgnoreCase("development")) {
```

Comment:
boolean exists = (existSchema != null) && (existSchema .getSummary() != null);

Reply:
if schema content is null, then local registry impl works well?

Reply:
why CI passed for preview version?

Reply:
1、has changed, but old version' summary content is null too, may has some potential problem
2、local impl work well, because it generate one list on startup
3、i don't know why CI can't check this error out. will check go on

Reply:
1.must compatible with old version
   treat summary is null as schema not registered, register again with summary, SC allow this action.
2.currently getSchemas of local impl is not equals to SC, infact it's a bug
3.ok, remember to check it.